dissect/target/tools/info.py

```diff
@@ -0,0 +1,145 @@
+        description="dissect.info",
+        fromfile_prefix_chars="@",
+        formatter_class=help_formatter,
+        add_help=False,
+    )
+    parser.add_argument("targets", metavar="TARGETS", nargs="*", help="Targets to display info from")
+    parser.add_argument("--from-file", nargs="?", help="file containing targets to load")
+    parser.add_argument("-d", "--delimiter", default=" ", action="store", metavar="','")
+    parser.add_argument("-s", "--strings", action="store_true", help="print output as string")
+    parser.add_argument("-r", "--record", action="store_true", help="print output as record")
+    parser.add_argument("-j", "--json-pretty", action="store_true", help="output records as pretty json")
+    parser.add_argument("-J", "--json", action="store_true", help="output records as one-line json")
+    configure_generic_arguments(parser)
+
+    args, rest = parser.parse_known_args()
+
+    if not args.targets and ("-h" in rest or "--help" in rest):
+        parser.print_help()
+        parser.exit()
+
+    process_generic_arguments(args)
+
+    if not args.targets and not args.from_file:
+        parser.error("too few arguments")
+
+    if args.from_file:
+        with open(args.from_file, "r") as f:
+            targets = f.read().splitlines()
+            while targets[-1] == "":
+                targets = targets[:-1]
+            args.targets = targets
+
+    for target in Target.open_all(args.targets):
+
+        if args.json:
+            print(json.dumps(obj_target_info(target)))
+        elif args.json_pretty:
+            print(json.dumps(obj_target_info(target), indent=4))
+        elif args.record:
+            rs = record_output(args.strings, args.json)
+            rs.write(
+                InfoRecord(
+                    os_family=target.os,
+                    os_version=target.version,
+                    ips=list(set(target.ips + target.ips_dhcp)),
+                    architecture=target.architecture,
+                    install_date=target.install_date,
+                    last_activity=target.activity,
+                    language=target.language,
+                    timezone=target.timezone,
+                    _target=target,
+                )
+            )
+        else:
+            print_target_info(target)
+
+
+def obj_target_info(target):
+    return {
+        "hostname": target.hostname,
+        "os_family": target.os,
+        "os_version": target.version,
+        "architecture": target.architecture,
+        "domain": target.domain,
+        "ips": list(set(target.ips + target.ips_dhcp)),
+        "install_date": str(target.install_date),
+        "last_activity": str(target.activity),
+        "language": target.language,
+        "timezone": target.timezone,
+        "disks": [{"type": d.__class__.__name__, "size": d.size} for d in target.disks],
+        "volumes": [{"name": v.name, "size": v.size, "fs": v.fs.__class__.__name__} for v in target.volumes],
```

Comment:
Filesystems optionally have `fs.__fstype__` that you can use, which will return a slug. E.g. `ntfs`.

Reply:
Good to know. Unfortunately for partitions/volumes with an unidentified filesystem (eg. `Microsoft reserved partition`), this wouldn't work as the returned class will be `NoneType`.

Reply:
Oops, good point. Then this is indeed the solution with the most straightfoward code.